code/modules/mining/machine_redemption.dm

```diff
@@ -179,6 +182,14 @@
 		if(user.transferItemToLoc(W, src))
 			inserted_disk = W
 			return TRUE
+			
+	if(istype(W, /obj/item/stack/ore))
```

Comment:
Now you have the same istype done twice.

Reply:
yes, it error'd out as just W, specifying it again passed compile check locally.  

`code/modules/mining/machine_redemption.dm:187:error: W.refined_type: undefined var`

Reply:
as an addendum, it was attack_hand that was ignoring the check before as well, if you throw a refined crystal on the floor, it gets recycled with no additions, but if you click on it, you bypass the auto grab check, this is why I added it here as well.

Reply:
You can just do
```
var/obj/item/stack/ore/O = W
if(istype(O)) do_stuff
```